test/smoke/testApps/HttpDependency/src/smokeTest/java/com/microsoft/applicationinsights/smoketest/HttpDependencyTest.java

```diff
@@ -0,0 +1,60 @@
+package com.microsoft.applicationinsights.smoketest;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+
+import java.util.List;
+
+import com.microsoft.applicationinsights.internal.schemav2.ExceptionData;
+import com.microsoft.applicationinsights.internal.schemav2.ExceptionDetails;
+import com.microsoft.applicationinsights.internal.schemav2.RemoteDependencyData;
+
+import org.junit.Test;
+
+@UseAgent
+public class HttpDependencyTest extends AiSmokeTest {
+
+    @Test
+    @TargetUri("/httpDependency")
+    public void testHttpDependency() {
+        assertEquals(1, mockedIngestion.getCountForType("RequestData"));
+
+        //Now only can get 3 remote dependency. Can not collect non-exist url failed http depnedncy if used Apache http client
+        //assertEquals(4, mockedIngestion.getCountForType("RemoteDependencyData"));
```

Comment:
Do you mean we do not collect failed dependencies when using ApacheHttpClient?

Reply:
I think that when giving a non-exist URL, it should produce a Dependency and an Exception.
But it only produce an exception using follow code:
"GetHttpUseApacheHttpClient("https://www.bingxxxxx.com")"

Reply:
I don't think so. In this case the dependency call doesn't complete itself and throws the exception in middle before the instrumentation call completes. So I think this is the right behavior. We should see failed HTTP dependency when we get something like 500 or similar result codes. @littleaj and @grlima can comment on this too.

Reply:
I don't know why using OKHttp Client can get ?

Reply:
In .NET SDK, we can collect failed http dependency if used non-exist URL.

Reply:
very interesting. @grlima are you aware about this behavior in our sdk?